docs/src/pages/components/selects/Listboxes.js

```diff
@@ -0,0 +1,159 @@
+import * as React from 'react';
+import clsx from 'clsx';
+import Box from '@mui/material/Box';
+import { useListbox } from '@mui/core/ListboxUnstyled';
+import { styled } from '@mui/system';
+
+const options = [
+  'Mercury',
+  'Venus',
+  'Earth',
+  'Mars',
+  'Jupiter',
+  'Saturn',
+  'Uranus',
+  'Neptune',
+  'Pluto',
+];
+
+const List = styled('ul')`
+  box-shadow: 0 4px 6px -2px rgba(0, 0, 0, 0.15);
+  padding: 0;
+  margin: 20px 0;
+
+  > li {
+    list-style: none;
+    padding: 4px 12px;
+    margin: 0;
+    min-width: 150px;
+    color: rgba(0, 0, 0, 0.85);
+    cursor: default;
+
+    &.highlighted {
+      background-color: #cde;
+      border-left: 4px solid #567;
+      padding-left: 8px;
+    }
+
+    &:hover {
+      background-color: #cde;
+    }
+
+    &.selected {
+      background-color: #345;
+      color: #fff;
+      font-weight: bold;
+    }
+
+    &.disabled {
+      opacity: 0.5;
+    }
+  }
+`;
+
+function renderOption(optionState, optionProps) {
+  const { selected, disabled, highlighted, option } = optionState;
+  const classes = {
+    selected,
+    disabled,
+    highlighted,
+  };
+
+  return (
+    <li key={option} {...optionProps} className={clsx(classes)}>
+      {option}
+    </li>
+  );
+}
+
+function CustomListbox(props) {
+  const listRef = React.useRef(null);
+
+  const { getRootProps, getOptionProps, getOptionState } = useListbox(
+    props,
+    listRef,
+  );
+
+  return (
+    <List {...getRootProps()} ref={listRef}>
+      {options.map((option, index) =>
+        renderOption(getOptionState(index), getOptionProps(index)),
+      )}
+    </List>
+  );
+}
+
+// allows to select at most two items
+
+const customReducer = (state, action, internalReducer) => {
```

Comment:
Yeah, it would work too. 

I thought it may be simpler for developers to just call a provided function if they want to use the default behavior instead of importing the default reducer. Why do you think import would work better here?

Reply:
For people used to the `reducer` signature, it would be the same, no need to learn new API. Importing function and calling it is also something people are already doing :)

Reply:
I haven't suggested people can't import a module :) I guess it's just a matter of discoverability - they would have to know that the default reducer exists and it's available for import. A function argument is just sitting there, waiting to be called.
I'm not precious about it, though, so I changed it as per your suggestion.